rasa/core/events/__init__.py

```diff
@@ -326,17 +347,7 @@ class BotUttered(Event):
     def __init__(self, text=None, data=None, metadata=None, timestamp=None):
         self.text = text
         self.data = data or {}
-        self._metadata = metadata or {}
-        super().__init__(timestamp)
-
-    @property
-    def metadata(self):
-        # needed for backwards compatibility <1.0.0 - previously pickled events
-        # won't have the `_metadata` attribute
-        if hasattr(self, "_metadata"):
-            return self._metadata
-        else:
-            return {}
+        super().__init__(timestamp, metadata)
```

Comment:
is there actually an official suggestion when to call the super-class constructor? In `java` you're forced to do it right at the start of the constructor? Just asking out of curiosity.

Reply:
In Java and C++, objects are built "from the inside out", so it makes sense to first call the super-constructor, so that the base object is built first. In Python however we're technically not building anything but just initializing, so we can choose the order we want. IMO it makes more sense to call the super-constructor first so that you can then use the members of the super class in the subclass constructor.

Reply:
>  IMO it makes more sense to call the super-constructor first so that you can then use the members of the super class in the subclass constructor.

I completely agree 👍